src/java.base/share/classes/jdk/internal/vm/vector/VectorSupport.java

```diff
@@ -294,42 +344,221 @@ public static long multiFieldOffset(Class<? extends VectorPayloadMF> cls) {
         }
     }
 
-    public primitive static class VectorPayloadMF64 extends VectorPayloadMF {
+    public primitive static class VectorPayloadMF64B extends VectorPayloadMF {
         @MultiField(value = 8)
         byte mfield = 0;
-        static long MFOFFSET = multiFieldOffset(VectorPayloadMF64.class);
+        static long MFOFFSET = multiFieldOffset(VectorPayloadMF64B.class);
```

Comment:
Any reason that we have to define the basic elment type to `mfield` for different type of VectorPayloadMF? We can do the `byte` type vector operations in compiler for multi-fields.

Reply:
This is done to streamline vector IR type creation and prevent adding special handling in compiler to append a reinterpret IR post LoadVector. I wanted to prevent introducing any special handling in C2 which does not comply with Java code e.g. Float256Vector is backed by VectorPayload256MF @multifield(value = 64) byte mfield;.

Reply:
OK, make sense to me. Per my understanding, it currently only has `load/store/zero` vector operations for the multi-fields, right? Hence always using byte vector operations may also work well to me. But making the vector type accurate with the concrete vector class type is better for compiler.  So this change is fine to me. Thanks!